src/shared/components/query/QueryStore.ts

```diff
@@ -143,46 +145,35 @@ export class QueryStore
 	}
 
 	@observable studiesHaveChangedSinceInitialization:boolean = false;
-	@observable savedVirtualCohorts:VirtualCohort[] = [];
+	@observable userVirtualStudies:VirtualStudy[] = [];
+	@observable deletedVirtualStudies:string[] = [];
 
 	@action public deleteVirtualCohort(id:string) {
-		VirtualCohorts.delete(id);
+		sessionServiceClient.deleteVirtualStudy(id)
+		.then(() => {
+			this.deletedVirtualStudies.push(id)
+		});
+	}
 
-		this.loadSavedVirtualCohorts();
+	@action public addVirtualCohort(id:string) {
+		sessionServiceClient.addVirtualStudy(id)
+		.then(() => {
+			this.deletedVirtualStudies = this.deletedVirtualStudies.filter(x => (x !== id));
+		});
 	}
 
 	@action private loadSavedVirtualCohorts() {
-		let localStorageVirtualCohorts:LocalStorageVirtualCohort[] = VirtualCohorts.get();
-		this.savedVirtualCohorts = localStorageVirtualCohorts.map((x:any)=>{
-			let samples:{studyId:string, sampleId:string}[] = [];
-			const constituentStudyIds:string[] = [];
-			for (const selectedCasesObj of x.selectedCases) {
-				samples = samples.concat(selectedCasesObj.samples.map((sampleId:string)=>({studyId:selectedCasesObj.studyID, sampleId})));
-				constituentStudyIds.push(selectedCasesObj.studyID);
-			}
-			return {
-				id: x.virtualCohortID,
-				name: x.studyName,
-				description: x.description,
-				samples,
-				constituentStudyIds
-			};
+		sessionServiceClient.getUserVirtualStudies().then((response) => {
+			this.userVirtualStudies = response;
 		});
 	}
 
-	@computed get virtualCohorts():VirtualCohort[] {
-		const ret:VirtualCohort[] = [];
-		// if (this.temporaryVirtualCohort.result) {
-		// 	ret.push(this.temporaryVirtualCohort.result);
-		// }
-		// for (let i=0; i<this.savedVirtualCohorts.length; i++) {
-		// 	ret.push(this.savedVirtualCohorts[i]);
-		// }
-		return ret;
+	@computed get virtualCohorts():VirtualStudy[] {

```

Comment:
maybe we get rid of this getter since we're just accessing another property?  we certainly don't need computed decorator here.